### ISSUE TITLE
Dynamic api (definition) update without app restart

### ISSUE THREAD

**Issue by alexeckert** [tiangolo/fastapi#2674]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [ ] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [ ] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [ ] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [ ] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

Hello,

thanks for creating FastAPI. I'm currently evaluating FastAPI for a specific use case where I want to create parts of the api definition dynamically. This seems to be unproblematic as long as I'm creating the pydantic models dynamically on app start, but I'm wondering if there is also the possibility to somehow dynamically exchange parts of api definition (e.g. triggered through some reload endpoint) without restarting the app and if and where such a mechanism could be implemented. Any hints?

**Comment by Mause** [comment]
Are you referring specifically to the documentation being updated as you mutate the API?

**Comment by alexeckert** [comment]
> 
> 
> Are you referring specifically to the documentation being updated as you mutate the API?

Hello @Mause, not specifically e.g. the data validation part also should be updated but the changes should be reflected in the documentation as well. So the question is if pydantic model updates (e.g. exchanging the class of a pydantic field) or model replacements somehow could be propagated to fastapi to update the served apis with the new data model (e.g. changed request body), while the endpoint (/myendpoint) is already defined.

**Comment by Mause** [comment]
I don't think that would be easily possible? Maybe if we understood your actual usecase we could better suggest a solution/path to take?

**Comment by tiangolo** [comment]
Thanks for the help here @Mause ! 👏  🙇 

If you solved the original problem, then you can close this issue @alexeckert  ✔️

...otherwise, please provide more info about your use case and what you are trying to solve, it's a bit abstract and I'm not sure I'm understanding what you want and how it would be achieved. Probably better if you can share some code with what you are trying to do.

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.

**Comment by devesh23** [comment]
@tiangolo : I have a similar request. The use case is explained below in simple bullet points. 

- I am trying to build a shared services framework. 
- Within this shared services framework, I want to have few core API's always available. 
- I want to be able to plug custom api during runtime. For example, Lets say I have a core animal api and bird api. 
- Now, I want to introduce fish api but the definitions of fish api would be written separately and would be either plugged in to the application using python plugin architecture or uploaded as a file or pip installed dynamically within the docker container. 
- Is there a neat way to achieve this without resulting in server restart?

Another way to imagine this problem is to think of the core fastapi service as the operating system in which we are installing apps and servicing use cases using the new apps.
